rust/operator-binary/src/controller.rs

```diff
@@ -1043,17 +1080,22 @@ fn build_rolegroup_statefulset(
         ));
     }
 
+    let metadata = ObjectMetaBuilder::new()
+        .with_recommended_labels(build_recommended_labels(
+            trino,
+            &resolved_product_image.app_version_label,
+            &rolegroup_ref.role,
+            &rolegroup_ref.role_group,
+        ))
+        .context(MetadataBuildSnafu)?
+        .with_annotation(
+            Annotation::try_from(("kubectl.kubernetes.io/default-container", "trino"))
+                .context(AnnotationBuildSnafu)?,
+        )
+        .build();
+
     pod_builder
-        .metadata_builder(|m| {
-            m.with_recommended_labels(build_recommended_labels(
-                trino,
-                &resolved_product_image.app_version_label,
-                &rolegroup_ref.role,
-                &rolegroup_ref.role_group,
-            ));
-            // This is actually used by some kuttl tests (as they don't specify the container explicitly)
-            m.with_annotation("kubectl.kubernetes.io/default-container", "trino")
-        })
```

Comment:
Can we keep the comment for the kuttl tests?

Reply:
https://github.com/stackabletech/trino-operator/pull/522/commits/2fa075a939849106a01c9ce90f0010c980c20048